drivers/gnss/gnss_publish.c

```diff
@@ -4,15 +4,77 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include "gnss_dump.h"
 #include "gnss_publish.h"
 #include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
 #include <zephyr/sys/iterable_sections.h>
 
 static struct k_spinlock lock;
 
+#if CONFIG_GNSS_LOG_DUMP
+static char dump_buf[CONFIG_GNSS_LOG_DUMP_BUF_SIZE];
+#endif  /* CONFIG_GNSS_LOG_DUMP */
+
+#if CONFIG_GNSS_LOG_DUMP
+static void gnss_publish_data_log_dump(const struct device *dev, const struct gnss_data *data)
```

Comment:
How about registering this and the satellite one as a normal gnss callback? Would save all the else cases and the custom traps in the process functions.

Reply:
I like that, I will update it to just use the callbacks :)